src/index/Index.cpp

```diff
@@ -156,52 +85,20 @@ void Index::createFromNTriplesFile(const string& ntFile, bool allPermutations) {
   v.resize(size_t(last - v.begin()));
   LOG(INFO) << "Done: unique." << std::endl;
   LOG(INFO) << "Size after: " << v.size() << std::endl;
-  createPermutation(indexFilename + ".pso", v, _psoMeta, 1, 0, 2);
-  LOG(INFO) << "Sorting for POS permutation..." << std::endl;
-  stxxl::sort(begin(v), end(v), SortByPOS(), STXXL_MEMORY_TO_USE);
-  LOG(INFO) << "Sort done." << std::endl;
-  createPermutation(indexFilename + ".pos", v, _posMeta, 1, 2, 0);
+  createPermutation<IndexMetaDataHmap>(indexFilename + ".pso", v, 1, 0, 2);
```

Comment:
The thing here is that we have the additional call to unique. There are three options:

- leave as is
- additional Argument bool performUnique (default value false, true for POS)
- "Automatic" decision via if constexpr (this is the POS call).

since the first is not good for readability and the third is somewhat dark magic I would prefer the second option, do you agree?

Reply:
I have chosen the second variant.